src/box/lua/slab.cc

```diff
@@ -265,6 +275,137 @@ lbox_slab_check(MAYBE_UNUSED struct lua_State *L)
 	return 0;
 }
 
+/*
+ * Decodes and returns the XML document returned by malloc_info() as is.
+ *
+ * This is an internal function that isn't supposed to be called by users,
+ * but it may be useful for developers.
+ *
+ * Returns an empty table if malloc_info() isn't supported by the system.
+ */
+static int
+lbox_malloc_internal_info(struct lua_State *L)
+{
+#if defined(HAVE_MALLOC_INFO)
+	char *buf;
+	size_t buf_size;
+	FILE *fp = open_memstream(&buf, &buf_size);
+	if (fp == NULL) {
+		diag_set(SystemError, "failed to open memory stream");
+		return luaT_error(L);
+	}
+	if (malloc_info(0, fp) != 0) {
+		diag_set(SystemError, "failed to get malloc info");
+		return luaT_error(L);
```

Comment:
You should `fclose` and `free` before returning.

Reply:
Sure. Fixed.

```diff
diff --git a/src/box/lua/slab.cc b/src/box/lua/slab.cc
index 63c7f4c78e2c..9d9e5e92e1c5 100644
--- a/src/box/lua/slab.cc
+++ b/src/box/lua/slab.cc
@@ -287,8 +287,8 @@ static int
 lbox_malloc_internal_info(struct lua_State *L)
 {
 #if defined(HAVE_MALLOC_INFO)
-	char *buf;
-	size_t buf_size;
+	char *buf = NULL;
+	size_t buf_size = 0;
 	FILE *fp = open_memstream(&buf, &buf_size);
 	if (fp == NULL) {
 		diag_set(SystemError, "failed to open memory stream");
@@ -296,6 +296,8 @@ lbox_malloc_internal_info(struct lua_State *L)
 	}
 	if (malloc_info(0, fp) != 0) {
 		diag_set(SystemError, "failed to get malloc info");
+		fclose(fp);
+		free(buf);
 		return luaT_error(L);
 	}
 	fclose(fp);
```